File: api/Bookshop.Application/Features/Customers/Commands/EditPassword/EditPasswordDto.cs
using System.Text.Json.Serialization;

namespace Bookshop.Application.Features.Customers.Commands.EditPassword
{
    public class EditPasswordDto
    {
        public string? Password { get; set; }
        public string? ConfirmPassword { get; set; }
        public string? NewPassword { get; set; }
        public string? ConfirmNewPassword { get; set; }
        [JsonIgnore]
        public string? UserId { get; set; }
    }
}

File: ui/src/app/dto/customer/edit-password-dto.ts

export class EditPasswordDto {
    password: string = '';
    confirmPassword: string = '';
    newPassword: string = '';
    confirmNewPassword: string = '';

    constructor(data?: Partial<EditPasswordDto>) {
        // Initialize properties with default values if not provided
        this.password = data?.password ?? this.password;
        this.confirmPassword = data?.confirmPassword ?? this.confirmPassword;
        this.newPassword = data?.newPassword ?? this.newPassword;
        this.confirmNewPassword = data?.confirmNewPassword ?? this.confirmNewPassword;
    }
}